src/EntityComponentManager_TEST.cc

```diff
@@ -2263,6 +2263,86 @@ TEST_P(EntityComponentManagerFixture, SerializedStateMsgAfterRemoveComponent)
   }
 }
 
+//////////////////////////////////////////////////
+// Verify SerializedStateMap message with no changed components,
+// but some removed components
+TEST_P(EntityComponentManagerFixture, SerializedStateMapMsgCompsRemovedOnly)
+{
+  // Create entity
+  Entity e1 = manager.CreateEntity();
+  auto e1c0 =
+    manager.CreateComponent<IntComponent>(e1, IntComponent(123));
+  manager.CreateComponent<DoubleComponent>(e1, DoubleComponent(0.0));
+  auto e1c2 =
+    manager.CreateComponent<StringComponent>(e1, StringComponent("int"));
+
+  manager.RunSetAllComponentsUnchanged();
+  manager.RemoveComponent(e1, e1c0);
+  manager.RemoveComponent(e1, e1c2);
+  // Serialize into a message
+  msgs::SerializedStateMap stateMsg;
+  manager.State(stateMsg);
+
+  // Check message
+  {
+    auto iter = stateMsg.entities().find(e1);
+    const auto &e1Msg = iter->second;
+    auto compIter = e1Msg.components().begin();
+
+    // Check number of components
+    EXPECT_EQ(e1Msg.components().size(), 2u);
```

Comment:
It's better to use `ASSERT_EQ` here to ensure the rest of the code doesn't run if the expected number of components are not in the container.